stock_picking_inter_warehouse/models/stock_warehouse.py

```diff
@@ -0,0 +1,86 @@
+# Copyright 2023 Ooops404
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, fields, models
+
+
+class StockWarehouse(models.Model):
+    _inherit = "stock.warehouse"
+
+    inter_warehouse_transfers = fields.Boolean("Inter-Warehouse Transfers")
+
+    receipt_destination_location_id = fields.Many2one(
+        "stock.location", "Receipt Destination Location"
+    )
+    receipt_picking_type_id = fields.Many2one(
+        "stock.picking.type", "Receipt Operation Type"
+    )
+    receipt_picking_partner_id = fields.Many2one("res.partner", "Partner in Picking")
+
+    def write(self, vals):
```

Comment:
@renda-dev Why not using standard methods provided by warehouse model in order to manage the activation of inter warehouses option ?

https://github.com/odoo/odoo/blob/ed8d82227b4a399798431d826f6fd013d2d0b888/addons/stock/models/stock_warehouse.py#L305

Reply:
Well, we are not really working around picking types here, and I don't see how hooking to that method could be useful in this case.

I'm totally open to suggestion on how can I improve this PR, please let me know what you think

Reply:
Okay so, I've spent quite some time looking into all of the methods that are available in `stock.warehouse` and haven't found any way to hook up to them.

In the meantime, I've improved this module's flexibility and performance.

Please let me know what you think and tell me if I'm missing something that I could do.

@rousseldenis

Reply:
@rousseldenis ping :)

Reply:
@rousseldenis gentle reminder (we need this in production asap - thanks!)